Detectors/TRD/workflow/src/TRDGlobalTrackingSpec.cxx

```diff
@@ -370,6 +371,43 @@ void TRDGlobalTracking::run(ProcessingContext& pc)
   std::iota(trackIdxArray.begin(), trackIdxArray.end(), 0);
   std::sort(trackIdxArray.begin(), trackIdxArray.end(), [tracksOutRaw](int lhs, int rhs) { return tracksOutRaw[lhs].getCollisionId() < tracksOutRaw[rhs].getCollisionId(); });
 
+  std::vector<std::pair<uint8_t, uint8_t>> pileUpDist;
+  bool ft0Seen = false;
+  if (mTrkMask[GTrackID::FT0]) { // pile-up tagging was requested
+    long maxDiffFwd = mTracker->Param().rec.trd.pileupFwdNBC;
+    long maxDiffBwd = mTracker->Param().rec.trd.pileupBwdNBC;
+    auto ft0recPoints = inputTracks.getFT0RecPoints();
+    auto trdTriggers = tmpInputContainer->mTriggerRecords;
+    ft0Seen = ft0recPoints.size() > 0;
+    pileUpDist.resize(trdTriggers.size(), {0, 0});
+    size_t curFT0 = 0;
+    for (size_t itrd = 0; itrd < trdTriggers.size(); itrd++) {
+      const auto& trig = trdTriggers[itrd];
+      uint8_t fwd = 0, bwd = 0;
+      for (size_t ft0id = curFT0; ft0id < ft0recPoints.size(); ft0id++) {
+        const auto& f0rec = ft0recPoints[ft0id];
+        if (f0rec.getTrigger().getVertex()) {
+          auto bcdiff = trig.getBCData().toLong() - f0rec.getInteractionRecord().toLong();
+          if (bcdiff > maxDiffBwd) {
+            curFT0 = ft0id + 1;
+            continue;
+          }
+          if (bcdiff > 0) { // pre-trigger pileup, maxDiffBwd is guaranteed to be < max uint8_t
+            if (bwd == 0) {
+              bwd = uint8_t(bcdiff);
+            }
+          } else {
+            if (bcdiff < -maxDiffFwd) {
+              break;
+            }
+            fwd = uint8_t(-bcdiff); // post-trigger pileup, maxDiffFwd is guaranteed to be < max uint8_t
```

Comment:
For the post-trigger collisions we should have the above check `if (fwd == 0)` and set it only once, no?

Reply:
Also, here and in some other places the code checker complains about missing parentheses for the casting, so you probably need either `(uint8_t)(...)` or `static_cast<uint8_t>(...)`

Reply:
> For the post-trigger collisions we should have the above check `if (fwd == 0)` and set it only once, no?

Why? Like above, since the FT0 is ordered and we want to find the furthest acceptable post-trigger collision, I keep updating fwd until the FT0 exceeds the acceptable range.

Reply:
> 

Looks like it did not line uintXXX_t times, changed to `unsigned char/short`.

Reply:
Right, sorry, for the time error we are interested in the furthest and not the closest collisions in the window of course